### PR TITLE
Remove the "config.platform.php" option when installing the Symfony Demo app

### LINKED ISSUE #255: Remove the config.platform.php config after installing the Symfony Demo app
The `NewCommand` removes the `config.platform.php` Composer option after installing Symfony (look in the `updateComposerJson()` method).

We should do the same for the `DemoCommand` to avoid this problem when installing the Symfony Demo app.

### PR DESCRIPTION
**Pull request by javiereguiluz**
This fixes #255.

### CODE REVIEW

File: src/Symfony/Installer/DownloadCommand.php [hunk 1]
```diff
@@ -545,6 +545,107 @@ protected function getUrlContents($url)
...
+     * It saves the given PHP array as the project's Composer config. In addition
+     * to JSON-serializing the contents, it synchronizes the composer.lock file to
+     * avoid out-of-sync Composer errors.
+     *
+     * @param array $config
+     */
+    protected function saveProjectComposerConfig(array $config)
+    {
+        $composerJsonFilepath = $this->projectDir.'/composer.json';
+        $this->fs->dumpFile($composerJsonFilepath, json_encode($config, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES)."\n");
```
Comment (Pierstoval) [inline src/Symfony/Installer/DownloadCommand.php]: These JSON constants are available starting with PHP 5.4, so 5.3 users will get an error.
 
Maybe use the int flag manually : `128 | 64` and add a comment about the constants?

File: src/Symfony/Installer/DemoCommand.php [hunk 2]
```diff
@@ -155,6 +156,33 @@ private function displayInstallationResult()
     }
 
     /**
+     * Updates the composer.json file to provide better values for some of the
+     * default configuration values.
+     *
+     * @return $this
+     */
+    protected function updateComposerJson()
```
Comment (xabbuh) [inline src/Symfony/Installer/DemoCommand.php]: Why not `private`?

Comment (javiereguiluz) [inline src/Symfony/Installer/DemoCommand.php]: Good question! I've refactored everything to move this method to the abstract parent class.

**Comment by javiereguiluz** [comment]
This is now ready for the final review. Thanks!

**Pull request closed without merging**